Firestore/Source/API/FIRAggregateField.mm

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import "FIRAggregateField.h"
+
+#import "Firestore/Source/API/FIRAggregateField+Internal.h"
+#import "Firestore/Source/API/FIRFieldPath+Internal.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+#pragma mark - FIRAggregateField
+
+@interface FIRAggregateField ()
+@property(nonatomic, strong) FIRFieldPath *fieldPath;
+@property(nonatomic, readwrite) model::AggregateField::OpKind _op;
+- (instancetype)initWithFieldPath:(nullable FIRFieldPath *)fieldPath
+                           opKind:(model::AggregateField::OpKind)op;
+@end
+
+@implementation FIRAggregateField
+- (instancetype)initWithFieldPath:(nullable FIRFieldPath *)fieldPath
+                           opKind:(model::AggregateField::OpKind)op {
+  if (self = [super init]) {
+    self.fieldPath = fieldPath;
+    self._op = op;
+  }
+  return self;
+}
+
+@synthesize fieldPath;
+
+- (const std::string)name {
+  return model::AggregateField::OperatorKind([self _op]);
+}
+
+- (model::AggregateField)createInternalValue {
+  if (self.fieldPath != Nil) {
+    return model::AggregateField([self _op], [self createAlias], self.fieldPath.internalValue);
+  } else {
+    return model::AggregateField([self _op], [self createAlias]);
+  }
+}
+
+- (model::AggregateAlias)createAlias {
+  if (self.fieldPath != Nil) {
+    return model::AggregateAlias([self name] + std::string{"_"} +
+                                 self.fieldPath.internalValue.CanonicalString());
+  } else {
+    return model::AggregateAlias([self name]);
+  }
+}
+
++ (instancetype)aggregateFieldForCount NS_SWIFT_NAME(count()) {
```

Comment:
I'm actually not sure if `NS_SWIFT_*` is needed in `.mm` files? do you know? I don't see them with one exception.

Reply:
So, given that we have `+ (instancetype)aggregateFieldForCount NS_SWIFT_NAME(count());` in FirAggregateField.h, can we remove `NS_SWIFT_NAME(count())` here?

Reply:
Right. I missed a few instances of this. Did a search and cleaned those up in this last commit. Thanks.